airbyte-cdk/python/connector_builder/connector_builder_handler.py

```diff
@@ -2,19 +2,36 @@
 # Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 #
 
+import dataclasses
 from datetime import datetime
+from functools import partial
+from typing import Any, Mapping
 
-from airbyte_cdk.models import AirbyteMessage, AirbyteRecordMessage, Type
+from airbyte_cdk.models import AirbyteMessage, AirbyteRecordMessage, ConfiguredAirbyteCatalog
+from airbyte_cdk.models import Type
+from airbyte_cdk.models import Type as MessageType
+from airbyte_cdk.sources.declarative.declarative_source import DeclarativeSource
 from airbyte_cdk.sources.declarative.manifest_declarative_source import ManifestDeclarativeSource
 from airbyte_cdk.utils.traced_exception import AirbyteTracedException
+from connector_builder.message_grouper import MessageGrouper
 
 
 def list_streams() -> AirbyteMessage:
     raise NotImplementedError
 
 
-def stream_read() -> AirbyteMessage:
-    raise NotImplementedError
+def read_stream(source: DeclarativeSource, config: Mapping[str, Any], configured_catalog: ConfiguredAirbyteCatalog) -> AirbyteMessage:
+    command_config = config.get("__test_read_config")
+    max_pages_per_slice = command_config["max_pages_per_slice"]
+    max_slices = command_config["max_slices"]
+    max_records = command_config["max_records"]
```

Comment:
Is there a reason to set a default here? This will only get used by the Java-side of the connector builder so it should be sufficient to set the values there